src/OrchardCore.Modules/OrchardCore.ContentFields/Views/HtmlField-Wysiwyg.Edit.cshtml

```diff
@@ -57,6 +59,12 @@
         if (defaultButtons[6] !== "insertShortcode") {
             defaultButtons.splice(6, 0, "insertShortcode");
         }
+        defaultButtons.forEach(group => {
+            if (group.includes("undo") && group.includes("redo")) {
+                group[group.indexOf("undo")] = "historyUndo";
+                group[group.indexOf("redo")] = "historyRedo";
+            }
```

Comment:
Is this only for when undo/redo are added by some overriding the default config? If yes, please add a comment about it. Also for the `HtmlBodyPart` editor.

Reply:
As far as I know, undo/redo is always enabled by the default options. If the options are being overridden, the default isn't used at all. That's the case on the Trumbowyg editors, where I only set the default configuration template to the history plugin. This uses the same technique as for the shortcodes button.